docs/DeveloperGuide.md

```diff
@@ -207,6 +207,37 @@ In particular, TAchy has 2 features that "add" items to the app. Firstly, there
 #### Example invocation sequence for AddAssignmentCommand
 <puml src="diagrams/AddAssignmentSequenceDiagram.puml" alt="Interactions Inside the Logic Component for the `add_assignment` Command" />
 
+## Viewing a student's details
+This `view_student` command is a feature that allows the user to expand and view all of a student's details in the details panel, based on the student's index.
+Additional details such as a student's assignments and remarks, that are not visible in the list panel, will now also be visible in the details panel.
+
+#### Design Considerations:
+In order to select a student to be displayed on the detail's panel, the student has to be identified from the list of contacts.
+
+**Aspect: Identifying the student to be displayed on the panel:**
+* Alternative 1 (current choice): Identify student by the student index.
+    * Pros: Easy to implement, and less prone to Users making errors in command format.
+    * Cons: The student index in the list may change every time the list is filtered, which may be confusing for the user as he has to check for the student index every time he calls this command.
+
+* Alternative 2: Identify student by their name.
+    * Pros: User will not have to check for the student index in the list every time he wants to call this command.
+    * Cons: As TAchy allows multiple contacts to be saved with the same name, additional fields such as phone number will have to be used in conjunction with "name" in order to differentiate between them. This will make the implementation more complex.
```

Comment:
It is good to mention that TAchy allows more than one person to have the same name, which is why we are not using method 2.